src/cmd/singularity/actions_test.go

```diff
@@ -174,6 +178,44 @@ func testSTDINPipe(t *testing.T) {
 	}
 }
 
+// testRunFromURI tests min fuctionality for singularity run/exec URI://
+func testRunFromURI(t *testing.T) {
+	tests := []struct {
+		name   string
+		image  string
+		action string
+		argv   []string
+		execOpts
+		exit          int
+		expectSuccess bool
+	}{
+		// Run from supported URI's and check the runscript call works
+		{"RunFromDocker", "docker://godlovedc/lolcow", "run", []string{}, execOpts{}, 0, true},
+		{"RunFromLibrary", "library://sylabsed/examples/lolcow:latest", "run", []string{}, execOpts{}, 0, true},
+		{"RunFromShub", "shub://GodloveD/lolcow", "run", []string{}, execOpts{}, 0, true},
+		// exec from a supported URI's and check the exit code
+		{"trueDocker", "docker://busybox:latest", "exec", []string{"true"}, execOpts{}, 0, true},
```

Comment:
Would like to see at least one test against one of @dctrud 's whiteout file docker tests.  Maybe he can council on which one would be best.

Reply:
We already test @dctrud whiteout issue at 

https://github.com/sylabs/singularity/blob/master/src/cmd/singularity/docker_test.go#L51